### ISSUE TITLE
Default Zend_Translate cache config wipes Config manifest cache

### ISSUE THREAD

**Issue by nyeholt** [silverstripe/silverstripe-framework#2706]
`i18n::include_by_locale()` calls `$cache->clean(Zend_Cache::CLEANING_MODE_ALL);` if "flush" is set in URL params. The default configuration of this cache is 

```
Zend_Translate::setCache(
                SS_Cache::factory('i18n', 'Output', array('lifetime' => null, 'automatic_serialization' => true))
            );
```

which ends up pointing to the same silverstripe-cache/user/cache directory that the config manifest uses. The net effect is that the execution of dev/build?flush=1 will first clear out this folder and rebuild the config manifest as expected in `ConfigManifest::regenerate()`, but later when calling `Folder::populateDefaults()` the `_t()` function is triggered, which then triggers `i18n::include_by_locale()` which _deletes_ those just built files

Typically, a subsequent request will hit code that checks the following in `ConfigManifest`

```
        if (false === $this->variantKeySpec) {
            $this->regenerate($includeTests);
        }
```

which rebuilds everything again, but in some cases this doesn't execute or the timing of requests is out (haven't nailed down the exact reason yet) and things don't regenerate, subsequently causing a failure by

```
PHP Warning:  Invalid argument supplied for foreach() in framework/core/manifest/ConfigManifest.php on line 140
```

A workaround is to manually configure a cache for Zend_Translate that stores data in a separate location.

**Comment by kinglozzer** [comment]
Partially fixed (for backends that support tags) by #3558?

**Comment by chillu** [comment]
This uses `Symfony\Component\Config\ResourceCheckerConfigCache` now in 4.x, which has resolved this issue implicitly - see https://github.com/silverstripe/silverstripe-framework/issues/6221 for context
